### PR TITLE
Add option -o | --loop, add some build improvements

### PR DESCRIPTION
**Pull request by docelic**
- add option -o | --loop
- update README.md
- bump version to 0.3.0
- require Crystal 0.24.1
- binary depends on all source files, not just src/cry.cr
- do not make assumption about user's way to obtain increased privileges
- rebuild doesn't trigger shards install if lib/ exists

Option -o | --loop will continuously run editor and execute.
It can be combined with option -b 1 to keep working on the same content
and executing it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,12 +6,14 @@ install: build force_link
 
 build: $(OUT_DIR)/cry
 
-$(OUT_DIR)/cry: src/cry.cr
+$(OUT_DIR)/cry: src/** lib
```
Comment (elorest) [inline Makefile]: Why is this necessary?

File: src/cry/command.cr [hunk 2]
```diff
@@ -3,27 +3,41 @@ require "colorize"
...
+                  elsif back_i > 0
+                    Dir.glob("./tmp/*_console.cr").sort.reverse[back_i - 1]?
                   end
 
-      system("cp #{_filename} #{@filename}") if _filename
+      system("cp #{_filename} #{@filename}") if _filename && File.exists?(_filename)
+    end
+
+    def new_file
+      "./tmp/#{Time.now.epoch_ms}_console.cr"
```
Comment (elorest) [inline src/cry/command.cr]: Call this filename

```cr
def filename
     "./tmp/#{Time.now.epoch_ms}_console.cr"
end
```

File: src/cry/command.cr [hunk 3]
```diff
@@ -3,27 +3,41 @@ require "colorize"
...
+    end
+    def new_result_file
```
Comment (elorest) [inline src/cry/command.cr]: Call this result_filename

```cr
def result_filename
     "./tmp/#{Time.now.epoch_ms}_console_result.log"
end
```

File: src/cry/command.cr [hunk 4]
```diff
@@ -43,18 +57,29 @@ module Cry
...
-        else
-          prepare_file
-          system("#{options.editor} #{@filename}")
-        end
+        loop do
+          unless args.code.blank? || File.exists?(args.code)
+            File.write(@filename, "puts (#{args.code}).inspect")
+          else
+            prepare_file
+            system("#{options.editor} #{@filename}")
```
Comment (elorest) [inline src/cry/command.cr]: Everywhere @filename is called use `filename`

File: src/cry/command.cr [hunk 5]
```diff
@@ -43,18 +57,29 @@ module Cry
...
-        File.write(@filename.sub("console.cr", "console_result.log"), result) unless result.blank?
-        puts result
+          File.write(@result_filename, result) unless result.nil?
+          puts result
+
+          break unless args.loop?
+          puts "\nENTER to edit, q to quit"
+          input = gets
+          break if input=~ /^q/i
+          @filename = new_file
```
Comment (elorest) [inline src/cry/command.cr]: Just set `@filename_seed = Time.now.epoch_ms` instead of @filename and @result_filename

File: src/cry/version.cr [hunk 6]
```diff
@@ -1,3 +1,3 @@
 module Cry
-  VERSION = "0.2.0"
+  VERSION = "0.3.0"
```
Comment (elorest) [inline src/cry/version.cr]: Don't change this until we release.

File: Makefile (hunk 1 above)
Comment (docelic) [reply Makefile]: After the : come the dependencies.
The compiled binary 'cry' depends on its sources and the lib/ directory ("depends" meaning it needs to be recompiled if the resources it depends on change).
The src/** is needed because without it, if you changed src/cry/command.cr, 'make' wouldn't realized cry had to be recompiled (and would tell you "Nothing to be done for 'all'" when you called it).
lib/ is needed because the compilation cannot begin before lib/ being present. (And conveniently, if lib/ isn't present then make looks up the "lib" target and runs "shards" to create it, estabilshing the proper dependency chain going from the final deliverable ("cry" binary) to the components necessary for building.)

**Comment by elorest** [comment]
@docelic Ping!
